jaxlie/_se3.py

```diff
@@ -220,6 +213,52 @@ def adjoint(self) -> jax.Array:
             axis=-2,
         )
 
+    @override
+    def jlog(self) -> jax.Array:
+        # Reference:
+        # Equations (179a, 179b, 180) from Micro-Lie theory:
+        # > https://arxiv.org/pdf/1812.01537
+        # and the Jlog6 implementation in Pinocchio:
+        # > https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/namespacepinocchio.html#a82e7cb47ae721d4161bbb143590096c5
+
+        rotation = self.rotation()
+        translation = self.translation()
+
+        jlog_so3 = rotation.jlog()
+
+        w = rotation.log()
+        theta = jnp.linalg.norm(w)
+
+        t2 = theta * theta
+        tinv = 1 / theta
+        t2inv = tinv * tinv
+        st, ct = jnp.sin(theta), jnp.cos(theta)
+        inv_2_2ct = 1 / (2 * (1 - ct))
+
+        beta = jnp.where(theta < 1e-6,
+                         1 / 12 + t2 / 720,
+                         t2inv - st * tinv * inv_2_2ct)
+
+        beta_dot_over_theta = jnp.where(theta < 1e-6,
+                                        1 / 360,
+                                        -2 * t2inv * t2inv + (1 + st * tinv) * t2inv * inv_2_2ct)
```

Comment:
this looks like it can produce NaNs for reverse-mode autodiff, due to  https://jax.readthedocs.io/en/latest/faq.html#gradients-contain-nan-where-using-where

you can see how we fix it elsewhere: https://github.com/search?q=repo%3Abrentyi%2Fjaxlie%20shim&type=code